spec/ddtrace/buffer_spec.rb

```diff
@@ -309,7 +309,8 @@ def get_test_items(n = 1)
   describe '#push' do
     let(:output) { buffer.pop }
 
-    subject(:push) { threads.each(&:join) }
+    subject(:push) { wait_for_threads }
+    let(:wait_for_threads) { threads.each { |t| t.join(5000) } }
```

Comment:
I normally add a timeout of 5 seconds to anything in our test suite that should be immediate, but technically has to wait. I do this so that tests don't hang forever.

Reply:
Perhaps we could use `raise "..." unless t.join(5000)` to raise on timeout? If for some reason the threads don't finish, rather than having the test suite silently chug along, possibly leaking threads that make later tests flaky, we could raise an exception here, which would immediately be spotted as an issue.

Reply:
⬆️  I'm not sure if you saw this comment. It may be a bit too much to do in this PR (or you may think it's not worth it), but just wanted to call out so that we make an explicit decision :)

Reply:
I did miss this one, thanks for calling this out.